File: user-service/src/main/java/socialapp/userservice/repository/FriendshipRepository.java
package socialapp.userservice.repository;

import org.springframework.stereotype.Repository;
import socialapp.userservice.model.entity.Friendship;
import socialapp.userservice.repository.common.CommonRepository;

import java.util.Set;

@Repository
public interface FriendshipRepository extends CommonRepository<Friendship,Long> {

    @Override
    default Class<?> entityClass(){
        return Friendship.class;
    };

    Set<Friendship> findAllByReceiverIdOrSenderId(Long receiverId, Long senderId);

}

File: user-service/src/main/kotlin/socialapp/ktuserservice/repository/FriendshipRepository.kt
package socialapp.ktuserservice.repository

import org.springframework.stereotype.Repository
import socialapp.ktuserservice.model.entity.Friendship
import socialapp.ktuserservice.repository.common.CommonRepository

@Repository
interface FriendshipRepository : CommonRepository<Friendship,Long> {

    fun findAllByReceiverIdOrSenderId(receiverId: Long, senderId: Long): Set<Friendship>

    override fun entityClass(): Class<*> = Friendship::class.java
}